src/main/java/org/elasticsearch/common/rounding/TimeZoneRounding.java

```diff
@@ -119,25 +94,25 @@ public Rounding build() {
         }
     }
 
-    static class TimeTimeZoneRoundingFloor extends TimeZoneRounding {
+    static class TimeUnitRounding extends TimeZoneRounding {
 
         static final byte ID = 1;
 
         private DateTimeUnit unit;
         private DateTimeField field;
         private DurationField durationField;
-        private DateTimeZone preTz;
-        private DateTimeZone postTz;
+        private DateTimeZone timeZone;
+        private boolean isUtc;
 
-        TimeTimeZoneRoundingFloor() { // for serialization
+        TimeUnitRounding() { // for serialization
         }
 
-        TimeTimeZoneRoundingFloor(DateTimeUnit unit, DateTimeZone preTz, DateTimeZone postTz) {
+        TimeUnitRounding(DateTimeUnit unit, DateTimeZone timeZone) {
             this.unit = unit;
-            field = unit.field();
-            durationField = field.getDurationField();
-            this.preTz = preTz;
-            this.postTz = postTz;
+            this.field = unit.field();
+            this.durationField = field.getDurationField();
+            this.timeZone = timeZone;
+            this.isUtc = timeZone.equals(DateTimeZone.UTC);
```

Comment:
Not sure if we need this boolean to guard against some timezone conversions. When the tz is UTC, it looks like the conversion is going to be very cheap anyway (just substracting a 0 if I'm not mistaken). It would also make the impl simpler.